src/arrayMethodSort.js

```diff
@@ -4,8 +4,22 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = 0; j < this.length - 1 - i; j++) {
+        if (compareFunction) {
+          if (compareFunction(this[j], this[j + 1]) > 0) {
+            [this[j], this[j + 1]] = [this[j + 1], this[j]];
+          }
+        } else {
+          if (String(this[j]) > String(this[j + 1])) {
+            [this[j], this[j + 1]] = [this[j + 1], this[j]];
+          }
+        }
+      }
+    }
+
+    return this;
```

Comment:
While the bubble sort algorithm is correctly implemented, it's worth noting that bubble sort is not the most efficient sorting algorithm for larger datasets due to its O(n^2) complexity. Consider researching more efficient algorithms such as quicksort or mergesort for better performance on large arrays.